sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/GroupingTable.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.query;
+
+import com.azure.cosmos.implementation.Document;
+import com.azure.cosmos.implementation.query.aggregation.AggregateOperator;
+import com.azure.cosmos.implementation.routing.UInt128;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class GroupingTable {
+    private static final List<AggregateOperator> EmptyAggregateOperators = new ArrayList<>();
+
+    private final Map<UInt128, SingleGroupAggregator> table;
+    private final Map<String, AggregateOperator> groupByAliasToAggregateType;
+    private final List<String> orderedAliases;
```

Comment:
can any of the public APIs of `GroupingTable` be invoked on different email threads. if so you will have race condition on updating these internal states.

Reply:
grouping will be sequential after collecting records from all the producers, so I dont see issue of multiple threads here.